react/src/pages/datasets/DatasetInfoDialog.tsx

```diff
@@ -0,0 +1,235 @@
+    const classes = useStyles();
+    const labeledBy = "dataset-info-dialog-slide-title";
+
+    const [dataset, setDataset] = useState<Dataset>();
+    const [analyses, setAnalyses] = useState<Analysis[]>([]);
+    const [sample, setSample] = useState<Sample>();
+    const [moreDetails, setMoreDetails] = useState(false);
+    const [showAnalysis, setShowAnalysis] = useState<boolean[]>([]);
+
+    function clickAnalysis(index: number) {
+        try {
+            // toggle
+            setShowAnalysis(
+                showAnalysis.map((val, i) => index === i ? !val : val)
+            );
+        } catch (error) {
+            console.error(error);
+        }
+    }
+
+    useEffect(() => {
+        fetch('/api/datasets/'+dataset_id)
+        .then(response => response.json())
+        .then(data => {
+            let temp = Object.assign({}, { ...data, analyses: undefined, tissue_sample: undefined });
+            setDataset(temp as Dataset);
+            setAnalyses(data.analyses as Analysis[]);
+            setShowAnalysis((data.analyses as Analysis[]).map(val => false));
+            setSample(data.tissue_sample as Sample);
+        })
+        .catch(error => {console.error(error)});
+
+    }, [dataset_id]);
+
+    return (
+        <Dialog onClose={onClose} aria-labelledby={labeledBy} open={open} maxWidth='lg' fullWidth={true}>
+            <DialogHeader id={labeledBy} onClose={onClose}>
+                Details of Dataset ID {dataset_id}
+            </DialogHeader>
+            <DialogContent className={classes.datasetInfo} dividers>
+                <Paper className={classes.paper} elevation={2}>
+                    {dataset &&
+                    <>
+                    <Grid container spacing={2} justify="space-evenly" className={classes.grid}>
+                        <Grid item xs={6}>
+                            {getInfo('Dataset ID', dataset.dataset_id)}
+                            {getInfo('Dataset Type', dataset.dataset_type)}
+                            {getInfo('Participant Codename', dataset.participant_codename)}
+                            {getInfo('Family Codename', dataset.family_codename)}
+                            {getInfo('Tissue ID', dataset.tissue_sample_id)}
+                            {getInfo('Sequencing Centre', dataset.sequencing_centre)}
+                        </Grid>
+                        <Grid item xs={6}>
+                            {getInfo('Notes', dataset.notes)}
+                            {getInfo('Created', dataset.created)}
+                            {getInfo('Created By', dataset.created_by)}
+                            {getInfo('Updated', dataset.updated)}
+                            {getInfo('Updated By', dataset.updated_by)}
+                        </Grid>
+                    </Grid>
+                    <Collapse in={moreDetails}>
+                    <Grid container spacing={2} justify="space-evenly">
+                        <Grid item xs={6}>
```

Comment:
Maybe you could make this a separate component? It is used a few times in this file and also in the participant dialog.